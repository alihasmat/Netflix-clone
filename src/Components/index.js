export { default as Accordian } from "./Accordian/Accordian";
export { default as Card } from "./Card/Card";
export { default as Feature } from "./Feature/Feature";
export { default as Footer } from "./Footer/Footer";
export { default as Form } from "./Form/Form";
export { default as Header } from "./Header/Header";
export { default as Jumbotron } from "./Jumbotron/Jumbotron";
export { default as Loading } from "./Loading/Loading";
export { default as Player } from "./Player/Player";
export { default as Profiles } from "./Profiles/Profiles";
export { default as Search } from "./Search/Search";
