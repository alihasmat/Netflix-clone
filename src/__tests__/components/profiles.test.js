import { getByTestId, render } from "@testing-library/react";
import { Profiles } from "../../Components";

describe("<Profiles />", () => {
  it(`renders <Profiles /> with populated data`, () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's Watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/Hasmat.png"
              data-testid="profile-image"
            />
            <Profiles.Name>Hasmat</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText("Who's Watching?")).toBeTruthy();
    expect(getByTestId("profile-image")).toBeTruthy();
    expect(getByText("Hasmat")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it(`renders <Profile /> with mis image id`, () => {
    const { container, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's Watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-image-misc" />
            <Profiles.Name>Hasmat</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByTestId("profile-image-misc")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
