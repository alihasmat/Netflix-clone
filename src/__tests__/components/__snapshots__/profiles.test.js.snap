// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders <Profile /> with mis image id 1`] = `
<div
  class="style__Container-sc-dud3s6-0 kdmfjV"
>
  <h1
    class="style__Title-sc-dud3s6-1 dbKuYb"
  >
    Who's Watching?
  </h1>
  <ul
    class="style__List-sc-dud3s6-2 hXrzAq"
  >
    <li
      class="style__Item-sc-dud3s6-5 eGoyRQ"
    >
      <img
        class="style__Picture-sc-dud3s6-3 jHsiRq"
        data-testid="profile-image-misc"
        src="images/misc/loading.gif"
      />
      <p
        class="style__Name-sc-dud3s6-4 bLFXpV"
      >
        Hasmat
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders <Profiles /> with populated data 1`] = `
<div
  class="style__Container-sc-dud3s6-0 kdmfjV"
>
  <h1
    class="style__Title-sc-dud3s6-1 dbKuYb"
  >
    Who's Watching?
  </h1>
  <ul
    class="style__List-sc-dud3s6-2 hXrzAq"
  >
    <li
      class="style__Item-sc-dud3s6-5 eGoyRQ"
    >
      <img
        class="style__Picture-sc-dud3s6-3 jHsiRq"
        data-testid="profile-image"
        src="/images/users//images/Hasmat.png.png"
      />
      <p
        class="style__Name-sc-dud3s6-4 bLFXpV"
      >
        Hasmat
      </p>
    </li>
  </ul>
</div>
`;
